iml-agent/src/action_plugins/stratagem/action_warning.rs

```diff
@@ -81,3 +84,49 @@ pub fn write_records(
 
     Ok(())
 }
+
+pub fn read_mailbox(device: &str, mailbox: &str, out: impl io::Write) -> Result<(), ImlAgentError> {
+    let mntpt = liblustreapi::search_rootpath(&device).map_err(|e| {
+        log::error!("Failed to find rootpath({}) -> {:?}", device, e);
+        e
+    })?;
+
+    let query: Vec<(String, String)> = Vec::new();
+    let mut wtr = csv::Writer::from_writer(out);
+    let message_endpoint = env::MANAGER_URL.join("/mailbox/")?.join(mailbox)?;
+    let identity = crypto_client::get_id(&env::PFX)?;
+    let client = crypto_client::create_client(identity)?;
+
+    let (sender, recv) = channel();
+
+    // Spawn off an expensive computation
+    tokio::spawn(
```

Comment:
Not only does this part need to be within a future, the entirety of read_mailbox needs to be modeled in a non-blocking way with futures so we don't block the runtime for other parallel computations.